src/components/PostDetails.tsx

```diff
@@ -1,117 +1,136 @@
-import React from 'react';
+import React, { useMemo, useState } from 'react';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
+import { Comment } from '../types/Comment';
+import { Post } from '../types/Post';
+import { deleteComment } from '../utils/api';
+
+interface Props {
+  selectedPost: Post | null;
+  comments: Comment[] | null;
+  isCommentLoading: boolean;
+  setComments: React.Dispatch<React.SetStateAction<Comment[] | null>>;
+  commentsError: boolean
+}
+
+export const PostDetails: React.FC<Props> = ({
+  selectedPost,
+  comments,
+  isCommentLoading,
+  setComments,
+  commentsError,
+}) => {
+  const [isFormOpened, setIsFormOpened] = useState<boolean>(false);
+
+  useMemo(() => {
```

Comment:
You use useMemo instead of useEffect to reset the form state when a new post is selected. useMemo is for memoizing expensive calculations, useEffect is for side effects like resetting state.